tests/test_cloudevent_functions.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import pathlib
+
+import pytest
+
+from cloudevents.http import CloudEvent, from_http, to_binary_http, to_structured_http
+
+from functions_framework import LazyWSGIApp, create_app, exceptions
+
+TEST_FUNCTIONS_DIR = pathlib.Path(__file__).resolve().parent / "test_functions"
+
+# Python 3.5: ModuleNotFoundError does not exist
+try:
+    _ModuleNotFoundError = ModuleNotFoundError
+except:
+    _ModuleNotFoundError = ImportError
+
+
+@pytest.fixture
+def cloudevent_1_0():
+    attributes = {
+        "specversion": "1.0",
+        "id": "my-id",
+        "source": "from-galaxy-far-far-away",
+        "type": "cloudevent.greet.you",
+    }
+    data = {"name": "john"}
+    return CloudEvent(attributes, data)
+
+
+@pytest.fixture
+def cloudevent_0_3():
+    attributes = {
+        "id": "my-id",
+        "source": "from-galaxy-far-far-away",
+        "type": "cloudevent.greet.you",
+        "specversion": "0.3",
```

Comment:
I'm surprised we need to support Cloud Events 0.3.

Reply:
We still support 0.3 although functionally a user cannot tell the difference between creating a 1.0 event vs a 0.3 event. If anything this can show future ccontributions how to add tests for newer cloudevent versions.